sys/include/xtimer.h

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup  sys_xtimer Timers
+ * @ingroup   sys
+ * @brief     Provides a high level timer module to register
+ *            timers, get current system time, and let a thread sleep for
+ *            a certain amount of time.
+ *
+ * The implementation takes one low-level timer that is supposed to run at 1MHz
+ * speed and multiplexes it.
+ *
+ * Insertion and removal of timers has O(n) complexity with (n) being the
+ * number of active timers.  The reason for this is that multiplexing is
```

Comment:
> "an upper bound", but any number of cycles between 0 and N, is not deterministic by this definition unless N is <= the margin of error. 

Actually, unless `N/2 < margin of error`, but I don't see how this contradicts my understanding.

Let's say setting an xtimer takes 10 cycles for 0 running timers, 12 cycles for 1 running timer and 30 cycles for 10 running timers. If I limit the number of running timers, I can guarantee that xtimer_set() will terminate within 20 ± 10 cycles (being 10 cycles the margin of error.